docs/chefs-external-api-configuration.md

```diff
@@ -0,0 +1,122 @@
+| userInfoEncryptionKey  | User Information Encryption Key       | encryption key supplied by Form Designer, stored encrypted in the db.                     |
+| userInfoEncryptionAlgo | User Information Encryption Algorithm | A CHEFS supported encryption algorithm.                                                   |
+| code                   | N/A                                   | Status Code. Only CHEFS Admin users can change this value. Used for the approval process. |
+
+**Recommendation** - secure the External API with an API Key and send user information in plain text.
+
+### User Info Object
+
+User (and form) context is provided to the External API in headers. This information can be encrypted and passed as a single header (the receiving API will decrypt) or plain text in multiple headers.
+The user information initially comes from the user's token, as such the values for each attribute may differ depending on which Identity Provider authenticated them.
+
+| Attribute    | Header                    | Purpose                                                            |
+| ------------ | ------------------------- | ------------------------------------------------------------------ |
+| formId       | X-CHEFS-FORM-FORMID       | the current CHEFS form id                                          |
+| versionId    | X-CHEFS-FORM-VERSIONID    | the version id of the form - optional, only populated if available |
+| submissionId | X-CHEFS-FORM-SUBMISSIONID | submission id - optional, only populated if available              |
+| userId       | X-CHEFS-USER-USERID       | User ID from the Identity Provider (ex. abcdef@idir)               |
+| username     | X-CHEFS-USER-USERNAME     | username/login name (ex. ASMITTY)                                  |
+| firstName    | X-CHEFS-USER-FIRSTNAME    | user's first name (ex. Alex)                                       |
+| lastName     | X-CHEFS-USER-LASTNAME     | user's last name (ex. Smitty)                                      |
+| fullName     | X-CHEFS-USER-FULLNAME     | user's Full name (ex. Smitty, Alex CITZ:EX)                        |
+| email        | X-CHEFS-USER-EMAIL        | user's email (ex. alex.smitty@gov.bc.ca)                           |
+| idp          | X-CHEFS-USER-IDP          | the Identity Provider code (ex. 'idir')                            |
+
+**Note** - Although the current user (the form submitter) token is available, there is no guarantee that it has not timed out. It will be stored upon entry/load of the form and is not refreshed.
+
+#### User Info Encryption
+
+The Form Designer will provide the encryption key for the selected encryption algorithm. This algorithm and key will encrypt the User Info into an encrypted value and passed in a single header (with header name = `userInfoHeader` value).
+
+The receiving api will have the same encryption key and algorithm and will decrypt on their end. They can then parse the decrypted user info object.
```

Comment:
Assuming that nobody is using (non-TLS) HTTP, is there a use case for this extra layer of encryption? Can't the third-party API encrypt how they want when they receive it?

Reply:
A bcgov api that would care about form id and user info and not have TLS is not likely to exist, true. But it’s entirely optional and costs us nothing to allow it. When messaging/events come online they will be using the same encryption methods at which point the consumer may find it easier to handle all data from CHEFS as encrypted. 

If you want to remove this and deal with it later (message queue) that’s fine and makes sense. Simple to remove this option.

Reply:
Yes it gets called on the designer. It’s actually kind of annoying. It’ll call once you put a url in. We will have to recommend setting all the other values first and set the url last. Not a huge deal but the proxy errors out without expected headers.